llvm/unittests/Target/AIE/AIEScheduleDAGMITest.cpp

```diff
@@ -0,0 +1,65 @@
+//===- MachineInstrTest.cpp -----------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+
+#include "AIEHazardRecognizer.h"
+#include "AIEMachineScheduler.h"
+#include "ScheduleDAGMITestUtils.h"
+
+using namespace llvm;
+
+namespace {
+
+class DummyAIEHazardRecognizer : public AIEHazardRecognizer {
+public:
+  DummyAIEHazardRecognizer(const ScheduleDAG *DAG)
+      : AIEHazardRecognizer(nullptr, nullptr, DAG, /*ScoreboardDepth=*/0) {}
+
+  ~DummyAIEHazardRecognizer() override {}
+
+  HazardType getHazardType(SUnit *SU, int DeltaCycles) override {
+    return NoHazard;
+  }
+};
+
+/// A simple wrapper around ScheduleDAGMITest that initializes a scheduler
+/// and a scheduling zone (SchedBoundary) with a DummyAIEHazardRecognizer.
+class AIEScheduleDAGMITest : public ScheduleDAGMITest {
+protected:
+  void initializeScheduler(bool IsPreRA = false) override {
+    assert(IsPreRA == false);
+    ScheduleDAGMITest::initializeScheduler(IsPreRA);
+    Scheduler->getSchedZone().HazardRec =
+        new DummyAIEHazardRecognizer(Scheduler.get());
+  }
+};
+
+/// Case where all instructions are scheduled in the current cycle
+TEST_F(AIEScheduleDAGMITest, SchedNoDelta) {
+  auto *MI0 = appendPlainInstr();
+  auto *MI1 = appendPlainInstr();
+  auto *MI2 = appendPlainInstr();
+
+  initializeScheduler();
+  SchedBoundary &Bot = Scheduler->getSchedZone();
+
+  // Mark all instructions as available.
+  for (auto *MI : {MI0, MI1, MI2})
+    Bot.releaseNode(Scheduler->getSUnit(MI), /*ReadyCycle=*/0, false);
+
+  Scheduler->scheduleInstr(MI2, Bot);
+  Bot.bumpCycle(2);
```

Comment:
As we are bumping here, I don't think we are scheduling in the current cycle, as in the description.

Reply:
We are scheduling in `Bot.getCurrCycle()`. Maybe I should rephrase the comment into `scheduled in Bot.getCurrCycle()`?